kube-client/src/config/file_config.rs

```diff
@@ -25,12 +25,15 @@ pub struct Kubeconfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub preferences: Option<Preferences>,
     /// Referencable names to cluster configs
-    pub clusters: Vec<NamedCluster>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub clusters: Option<Vec<NamedCluster>>,
     /// Referencable names to user configs
     #[serde(rename = "users")]
-    pub auth_infos: Vec<NamedAuthInfo>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub auth_infos: Option<Vec<NamedAuthInfo>>,
```

Comment:
Does it make sense to use `Vec<X>` and `#[serde(default, skip_serializing_if = "Vec::is_empty")]` here rather than `Option<Vec<X>>`?

It looks like upstream heavily uses [omitempty with +optional](https://github.com/kubernetes/client-go/blob/7697067af71046b18e03dbda04e01a5bb17f9809/tools/clientcmd/api/types.go) meaning they do not distinguish between empty and not-present.

Reply:
Awesome, I didn't know that was a thing! I tried with default only and I couldn't make it work. The PR is a lot simpler now 😊

Reply:
Yeah, this is great now. Avoids the (minor) breaking config change also and can mark this as a fix.